war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -2923,18 +2923,24 @@ function applyErrorMessage(elt, rsp) {
     if (rsp.status == 200) {
         elt.innerHTML = rsp.responseText;
     } else {
-        var id = 'valerr' + (iota++);
-        elt.innerHTML = '<a href="" onclick="document.getElementById(\'' + id
-        + '\').style.display=\'block\';return false">ERROR</a><div id="'
-        + id + '" style="display:none">' + rsp.responseText + '</div>';
-        var error = document.getElementById('error-description'); // cf. oops.jelly
-        if (error) {
-            var div = document.getElementById(id);
-            while (div.firstElementChild) {
-                div.removeChild(div.firstElementChild);
-            }
-            div.appendChild(error);
-        }
+        var parser = new DOMParser();
+        var doc = parser.parseFromString(rsp.responseText, 'text/html');
+        var message = doc.querySelector('#error-description') || doc.body;
+        // Remove styles and scripts
+        var harmfulElements = doc.querySelectorAll('style, script');
+        harmfulElements.forEach(function(el) {
+            el.parentElement.removeChild(el);
+        });
+        // Clear out previous errors
+        while(elt.firstChild) {
+            elt.removeChild(elt.firstChild);
+        }
+        var details = document.createElement('details');
+        var summary = document.createElement('summary');
+        summary.textContent = 'Field Validation ERROR';
```

Comment:
I did opt to put in some more text to indicate what kind of error we’re dealing with. Not sure if i18n is required (or how that would be done).

Reply:
I now realize that “Field Validation ERROR” can be assumed to just be the same as a field that doesn’t match the validation constraints.

Maybe this is clearer:

```suggestion
        summary.textContent = 'Error validating this field';
```